dev/build_release.py

```diff
@@ -200,59 +179,57 @@ def start_deb_build(self, name):
+
+  @classmethod
+  def __debian_build(cls, name, options, build_number, gradle_root):
+    jarRepo = options.jar_repo
+    parts = options.bintray_repo.split('/')
     if len(parts) != 2:
       raise ValueError(
           'Expected --bintray_repo to be in the form <owner>/<repo>')
     org, packageRepo = parts[0], parts[1]
     bintray_key = os.environ['BINTRAY_KEY']
     bintray_user = os.environ['BINTRAY_USER']
 
-    if self.__options.nebula:
+    if options.nebula:
       target = 'candidate'
       extra_args = [
-          '--stacktrace',
-          '--info',
-          '-Prelease.useLastTag=true',
-          '-PbintrayPackageBuildNumber={number}'.format(
-              number=self.__build_number),
-          '-PbintrayOrg="{org}"'.format(org=org),
-          '-PbintrayPackageRepo="{repo}"'.format(repo=packageRepo),
-          '-PbintrayJarRepo="{jarRepo}"'.format(jarRepo=jarRepo),
-          '-PbintrayKey="{key}"'.format(key=bintray_key),
-          '-PbintrayUser="{user}"'.format(user=bintray_user)
-        ]
+        '--stacktrace',
+        '-Prelease.useLastTag=true',
+        '-PbintrayPackageBuildNumber={number}'.format(number=build_number),
+        '-PbintrayOrg="{org}"'.format(org=org),
+        '-PbintrayPackageRepo="{repo}"'.format(repo=packageRepo),
+        '-PbintrayJarRepo="{jarRepo}"'.format(jarRepo=jarRepo),
+        '-PbintrayKey="{key}"'.format(key=bintray_key),
+        '-PbintrayUser="{user}"'.format(user=bintray_user)
+      ]
     else:
       target = 'buildDeb'
       extra_args = []
 
-    if self.__options.debug_gradle:
+
+    if options.info_gradle:
+      extra_args.append('--info')
+    if options.debug_gradle:
       extra_args.append('--debug')
 
-    if self.__options.gradle_cache_path:
-      extra_args.append('--gradle-user-home={}'.format(self.__options.gradle_cache_path))
+    if options.gradle_cache_path:
+      extra_args.append('--gradle-user-home={}'.format(options.gradle_cache_path))
 
     if name == 'deck' and not 'CHROME_BIN' in os.environ:
       extra_args.append('-PskipTests')
     elif name == 'halyard':
       extra_args.append('-PbintrayPackageDebDistribution=trusty-nightly')
 
-    # Currently spinnaker is in a separate location
-    gradle_root = self.determine_gradle_root(name)
     print 'Building and publishing Debian for {name}...'.format(name=name)
-    # Note: 'candidate' is just the gradle task name. It doesn't indicate
-    # 'release candidate' status for the artifacts created through this build.
-    return BackgroundProcess.spawn(
-      'Building and publishing Debian for {name}...'.format(name=name),
-      'cd "{gradle_root}"; ./gradlew {extra} {target}'.format(
-          gradle_root=gradle_root, extra=' '.join(extra_args), target=target),
-      logfile='{name}-debian-build.log'.format(name=name)
-    )
+    cmds = [
+      './gradlew {extra} {target}'.format(extra=' '.join(extra_args), target=target)
+    ]
+    logfile = '{name}-debian-build.log'.format(name=name)
```

Comment:
do you want to delete the existing one since you append? Dont rely on the caller/jenkins to do that.